examples/seismic/elastic/wavesolver.py

```diff
@@ -93,6 +106,59 @@ def forward(self, src=None, rec1=None, rec2=None, v=None, tau=None,
         kwargs.update(model.physical_params(**kwargs))
 
         # Execute operator and return wavefield and receiver data
-        summary = self.op_fwd(save).apply(src=src, rec1=rec1, rec2=rec2,
-                                          dt=kwargs.pop('dt', self.dt), **kwargs)
-        return rec1, rec2, v, tau, summary
+        summary = self.op_fwd(save).apply(src=src, rec_tau=rec_tau, rec_vx=rec_vx,
+                                          rec_vz=rec_vz, dt=kwargs.pop('dt', self.dt),
+                                          **kwargs)
+        if self.model.grid.dim == 3:
+            return rec_tau, rec_vx, rec_vz, v, tau, summary, rec_vy
+        return rec_tau, rec_vx, rec_vz, v, tau, summary
```

Comment:
thios change of order with a rec at the end in 3d isn't very good

Reply:
agreed.

Perhaps we could return a `namedtuple` instead? or just None... so that we always return the same number of arguments, in a given order